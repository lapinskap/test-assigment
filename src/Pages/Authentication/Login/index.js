import React, { useContext } from 'react';
import {
  Col,
} from 'reactstrap';
import { useLocation } from 'react-router-dom';
import RbsContext from '../../../utils/RoleBasedSecurity/RbsContext';
import Tiles from '../../../Components/Tiles';

const Login = () => {
  const rbsContext = useContext(RbsContext);
  const location = useLocation();

  const login = async (type) => {
    await rbsContext.login(type);
  };

  const referer = location.state && location.state.referer ? location.state.referer : null;
  return (
    <div className="h-100 bg-white bg-animation">
      <div className="d-flex h-100 justify-content-center align-items-center">
        <Col className="mx-auto app-login-box">
          <div className="app-logo-inverse mx-auto mb-3" />
          <div className="mx-auto card" style={{ maxWidth: '700px' }}>
            <div className="card-body">
              <Tiles
                selected={null}
                onTileClick={(id) => login(id)}
                md="6"
                sm="6"
                config={[
                  {
                    label: 'OMB mock',
                    icon: 'pe-7s-user',
                    id: 'omb',
                  },
                  {
                    label: 'AHR mock',
                    icon: 'pe-7s-user',
                    id: 'ahr',
                  },
                  {
                    label: 'OMB Prawdziwe dane',
                    icon: 'pe-7s-user',
                    id: 'admin',
                  },
                  {
                    label: 'AHR Prawdziwe dane',
                    icon: 'pe-7s-user',
                    id: 'adminahr',
                  },
                ]}
              />
            </div>
            <div className="modal-footer clearfix">
              <div className="float-left">
                <a
                  href={`/signin/oauth/client/login${referer ? `?referer_url=${encodeURIComponent(`/#${referer}`)}` : ''}`}
                  color="link"
                  className="btn-lg btn btn-link"
                >
                  Zaloguj z MyBenefit
                </a>
              </div>
            </div>
          </div>

          <div className="text-center opacity-8 mt-3">
            Copyright &copy; MyBenefit
            {' '}
            {new Date().getFullYear()}
          </div>
        </Col>
      </div>
    </div>
  );
};

export default Login;
