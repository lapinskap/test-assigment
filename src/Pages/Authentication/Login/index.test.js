import React from 'react';
import renderer from 'react-test-renderer';
import Login from './index';
import '../../../tests/setupTests';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/example/path',
  }),
  // eslint-disable-next-line react/prop-types,react/jsx-props-no-spreading
  Link: ({ to, children, ...props }) => <a href={to} {...props}>{children}</a>,
}));
jest.mock('reactstrap', () => ({
  ...jest.requireActual('reactstrap'),
  Tooltip: () => <div>Tooltip mock</div>,
}));
describe('<Login />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render correct in default state', () => {
    const tree = renderer
      .create(<Login />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
