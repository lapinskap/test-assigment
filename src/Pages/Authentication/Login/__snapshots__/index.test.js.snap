// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Should render correct in default state 1`] = `
<div
  className="h-100 bg-white bg-animation"
>
  <div
    className="d-flex h-100 justify-content-center align-items-center"
  >
    <div
      className="mx-auto app-login-box col"
    >
      <div
        className="app-logo-inverse mx-auto mb-3"
      />
      <div
        className="mx-auto card"
        style={
          Object {
            "maxWidth": "700px",
          }
        }
      >
        <div
          className="card-body"
        >
          <div
            className="grid-menu grid-menu-4col"
          >
            <div
              className="no-gutters row"
              data-t1="tiles"
            >
              <div
                className="bg-white col-sm-6 col-md-6"
              >
                <div
                  className="widget-chart widget-chart-hover text-primary"
                  data-t1="omb"
                  onClick={[Function]}
                  role="presentation"
                >
                  <div
                    className="icon-wrapper rounded-circle"
                  >
                    <div
                      className="icon-wrapper-bg bg-primary"
                    />
                    <i
                      className="pe-7s-user"
                    />
                  </div>
                  <div
                    className="widget-numbers-sm"
                  >
                    OMB mock
                  </div>
                </div>
              </div>
              <div
                className="bg-white col-sm-6 col-md-6"
              >
                <div
                  className="widget-chart widget-chart-hover text-primary"
                  data-t1="ahr"
                  onClick={[Function]}
                  role="presentation"
                >
                  <div
                    className="icon-wrapper rounded-circle"
                  >
                    <div
                      className="icon-wrapper-bg bg-primary"
                    />
                    <i
                      className="pe-7s-user"
                    />
                  </div>
                  <div
                    className="widget-numbers-sm"
                  >
                    AHR mock
                  </div>
                </div>
              </div>
              <div
                className="bg-white col-sm-6 col-md-6"
              >
                <div
                  className="widget-chart widget-chart-hover text-primary"
                  data-t1="admin"
                  onClick={[Function]}
                  role="presentation"
                >
                  <div
                    className="icon-wrapper rounded-circle"
                  >
                    <div
                      className="icon-wrapper-bg bg-primary"
                    />
                    <i
                      className="pe-7s-user"
                    />
                  </div>
                  <div
                    className="widget-numbers-sm"
                  >
                    OMB Prawdziwe dane
                  </div>
                </div>
              </div>
              <div
                className="bg-white col-sm-6 col-md-6"
              >
                <div
                  className="widget-chart widget-chart-hover text-primary"
                  data-t1="adminahr"
                  onClick={[Function]}
                  role="presentation"
                >
                  <div
                    className="icon-wrapper rounded-circle"
                  >
                    <div
                      className="icon-wrapper-bg bg-primary"
                    />
                    <i
                      className="pe-7s-user"
                    />
                  </div>
                  <div
                    className="widget-numbers-sm"
                  >
                    AHR Prawdziwe dane
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal-footer clearfix"
        >
          <div
            className="float-left"
          >
            <a
              className="btn-lg btn btn-link"
              color="link"
              href="/signin/oauth/client/login"
            >
              Zaloguj z MyBenefit
            </a>
          </div>
        </div>
      </div>
      <div
        className="text-center opacity-8 mt-3"
      >
        Copyright Â© MyBenefit
         
        2021
      </div>
    </div>
  </div>
</div>
`;
